app/models/miq_search.rb

```diff
@@ -8,6 +8,16 @@ class MiqSearch < ApplicationRecord
 
   before_destroy :check_schedules_empty_on_destroy
 
+  # validate if the name of a new filter is unique in Global Filters
```

Comment:
Looking at this I was thinking that perhaps the same thing can be accomplished with a validation on `description` that has a condition on `search_type`. Maybe something like this -
```
  validates_uniqueness_of :description, :scope => "db", conditions: -> { where.not(:search_type => 'user') }
```
What do you think? It would probably force you to change the way it gets checked on the UI side. But, it would be much simpler.